File: python-version/LC1760/minimum_size.py
class Solution:
    def minimumSize(self, nums: list[int], maxOperations: int) -> int:
        mx = max(nums)
        low, high = 0, mx
        while low+1 < high:
            mid = (low+high)//2
            if self.check(nums, maxOperations, mid):
                high = mid
            else:
                low = mid
        return high
    
    def check(self, nums: list[int], maxOperations: int, mid: int) -> bool:
        cnt = 0
        for x in nums:
            cnt += (x-1)//mid
        return cnt <= maxOperations

File: go-version/leetcode/LC1760/minimum_size.go
package lc1760

func minimumSize(nums []int, maxOperations int) int {
	mx := 0
	for _, num := range nums {
		mx = max(mx, num)
	}

	low, high := 0, mx
	for low+1 < high {
		mid := (low + high) >> 1
		if check(nums, maxOperations, mid) {
			high = mid
		} else {
			low = mid
		}
	}
	return high
}

func check(nums []int, maxOperations, mid int) bool {
	var cnt int
	for _, num := range nums {
		cnt += (num - 1) / mid
	}
	return cnt <= maxOperations
}
